spec/defines/sslcertificate_spec.rb

```diff
@@ -60,46 +65,45 @@
   #  }}
   #
   #  it { expect { should contain_exec('Install-SSL-Certificate-testCert')}.to raise_error(Puppet::Error, /Must pass password to Sslcertificate\[certificate-testCert\]/) }
-  #end
+  # end
 
   describe 'when no certificate location is provided' do
     let(:title) { 'certificate-testCert' }
     let(:params) { {
-        :name       => 'testCert',
-        :password   => 'testPass',
-        :thumbprint => '07E5C1AF7F5223CB975CC29B5455642F5570798B',
-        :root_store => 'LocalMachine',
-        :store_dir  => 'My',
+      :name       => 'testCert',
+      :password   => 'testPass',
+      :thumbprint => '07E5C1AF7F5223CB975CC29B5455642F5570798B',
+      :root_store => 'LocalMachine',
+      :store_dir  => 'My',
     }}
 
-    it { expect { should contain_exec('Install-SSL-Certificate-testCert')}.to raise_error(Puppet::Error, /Must pass location to Sslcertificate\[certificate-testCert\]/) }
+    it { expect { should contain_exec('Install-SSL-Certificate-testCert') }.to raise_error(Puppet::Error) }
```

Comment:
That's correct, that variable is required to be set for a defined type because no default is provided. I believe the exact wording of the error will be Puppet version dependent. For instance, on Puppet 4 we get:
```Failure/Error: it { expect { should contain_exec('Install-SSL-Certificate-testCert')}.to raise_error(Puppet::Error, /Must pass location to Sslcertificate\[certificate-testCert\]/) }
       expected Puppet::Error with message matching /Must pass location to Sslcertificate\[certificate-testCert\]/, got #<Puppet::PreformattedError: Evaluation Error: Error while evaluating a Resource Statement, Sslcertificate[certificate-testCert]: expects a value for parameter 'location' at line 1 on node pc```